### DISCUSSION TITLE
Expression language : Lint

### DISCUSSION THREAD

**Discussion by kl3sk** [symfony/symfony#58688] [Q&A] [answered]
Hello,

I'am currently playing with this component and want to make some basic arithmetics.

Because `division by 0` is not possible, I want to catch this error to prevent user. So I see the `lint` method, that check and return valid or not.

As the documentation says, `lint` can return `true` if expression is valid. But as I see in the [source](https://github.com/symfony/symfony/blob/6161368d792b72b313c25e4706ac1b09107f2193/src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php#L101) it doesn't return anything -> `void` return .

I see other problems in the doc. I may open an issue if i'am correct about this error.

Thanks

Edit:

References:
- https://symfony.com/doc/current/components/expression_language.html#parsing-and-linting-expressions
- https://github.com/symfony/symfony/blob/7.1/src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php#L101

**Comment by stof** [comment, marked as answer]
Dividing by 0 is a runtime error, not a syntax error. It depends on the values used during the evaluation of the expression (we cannot know at parse time whether `a / b` is a division by 0, it depends on the value of `b`)

> As the documentation says, `lint` can return `true` if expression is valid. But as I see in the [source](https://github.com/symfony/symfony/blob/6161368d792b72b313c25e4706ac1b09107f2193/src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php#L101) it doesn't return anything -> `void` return .

this is a bug in the documentation. Linting throws an exception for invalid expressions. It is not about the return value. Please open an issue on https://github.com/symfony/symfony-docs/ (or even better a PR with a fix)

**Reply by kl3sk** [reply]
Ok, thanks you,

I wished diving by zero could be catched. Anyway, no one should divide by zero 😋 

> this is a bug in the documentation. Linting throws an exception for invalid expressions. It is not about the return value

Yep, that what the code says.

**Reply by kl3sk** [reply]
@stof 

Here is a proposition: https://github.com/symfony/symfony-docs/pull/20353
